consensus/polybft/validators_snapshot.go

```diff
@@ -323,6 +327,12 @@ func (v *validatorsSnapshotCache) getLastCachedSnapshot(currentEpoch uint64,
 		}
 	}
 
+	if cachedSnapshot != nil && cachedSnapshot.Epoch > epochToQuery {
+		// if we have a snapshot in memory or db, but it's newer than the one we need
+		// then return nil, and just recalculate it from scratch
+		return nil, nil
+	}
```

Comment:
Do we still need this check? Seems like we don't, because you have changed the logic behind reading snapshot from the db and I don't see how we could read the snapshot from "the future".

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/164/commits/3307b87eccfa5b80c64502f37615cc60f6b4613e